### ISSUE TITLE
SSE for everything!

### ISSUE THREAD

**Issue by zee2200** [HandmadeMath/Handmade-Math#74]
your library's speed could easily be improved signficantly but having vec2 and vec3 be padded to be 16 bytes long, you then can use SSE for all of those calculations. Also consider directly passing __m128 instead of passing a union, that should also give you a performance boost, I'd love to go through and edit the library myself but I'm working on my own right now, jut thought I'd leave this suggestion here to share what I found.

**Comment by RandyGaul** [comment]
Directly passing around __m128, in my own tests, hasn't been necessary for quite a number of years now. At least for MSVC.

**Comment by zee2200** [comment]
Fair enough, I'm using gcc so that's probably the difference.

**Comment by bvisness** [comment]
Zak and I tried to SSE our vec2 and vec3 operations before, but padding them to 16 bytes breaks the union stuff we do with the bigger types (like [this](https://github.com/HandmadeMath/Handmade-Math/blob/master/HandmadeMath.h#L367), for example). Plus, given the library's focus on 3D graphics, we figured it was probably ok if vec2 and vec3 weren't explicitly SSE'd.

Really, though, Handmade Math has hardly any support for vec2 and vec3 at the moment, since we don't have mat2 or mat3 to complement them. Perhaps a future major version will try to handle them more gracefully.
